client/src/actions/scoreForm.js

```diff
@@ -174,42 +133,98 @@ export function activitiesSubmitData(id, index, source, date, initialValue, newV
 				return response;
 			})
 			.then((response) => response.json())
-			.then((activities) => dispatch(updateActivity(source, index, type, initialValue, newValue, true)))
-			.catch((err) => console.log(err))
-		};				
+				.then((response) => dispatch(activitiesFetchDataSuccess(response, source)))
+				.catch((err) => console.log(err))
+		};
 	}
-}
 
-export function activitiesListFetchData(date, source) {
-	return dispatch => {
-		fetch('/user/1/activitylist', {
+	export function activitiesSubmitData(id, index, source, date, initialValue, newValue, type) {
+		let amount = parseFloat(newValue)
+			if (isNaN(amount)) {
+				amount = 0
+			}
+		if (source === 'exercise') {
```

Comment:
It's almost entirely whitespace. If you use commandline and diff with the -w flag you can see the actual lines that were changed, which look to be:

 export function activityListSubmitData(activities, ids, date, action) {
        return dispatch => {
+               // TODO: this can be null, right?
+               // TODO: this should be an each, yeah? Test to make sure
+               //activities.each((activity, index) => {
                activities.map((activity, index) => {
                        let active = action === 'add' ? true : false
                                fetch('/user/1/activitylist/record', {
+                                       credentials: 'same-origin',
                                        method: 'post',
                                        headers: {
                                                "Content-Type": "application/json"
@@ -38,6 +42,7 @@ export function updateActivityList(date, activities, action, activity) {
                return dispatch => {
                        fetch('/user/1/activitylist', {
                                method: 'post',
+                               credentials: 'same-origin',
                                headers: {
                                        "Content-Type": "application/json"
                                },
@@ -109,10 +114,13 @@ export function activitiesFetchDataSuccess(activities, source) {
        export function activitiesFetchData(date, source) {
                return dispatch => {
                        fetch('/user/1/activity', {
+                               credentials: 'same-origin',
+
                                method: 'post',
                                headers: {
                                        "Content-Type": "application/json"
                                },
+
                                body: JSON.stringify({
                                        day: date,
                                        source: source
@@ -158,6 +166,7 @@ export function activitiesSubmitData(id, index, source, date, initialValue, newV
                        if (source === 'bonus' || initialValue !== amount) {
                                fetch('/user/1/activity/record', {
                                        method: 'post',
+                                       credentials: 'same-origin',
                                        headers: {
                                                "Content-Type": "application/json"
                                        },
@@ -184,6 +193,7 @@ export function activitiesListFetchData(date, source) {
                return dispatch => {
                        fetch('/user/1/activitylist', {
                                method: 'post',
+                               credentials: 'same-origin',
                                headers: {
                                        "Content-Type": "application/json"
                                },
@@ -200,7 +210,12 @@ export function activitiesListFetchData(date, source) {
                        })
                        .then((response) => response.json())
                                .then((response) => dispatch(activitiesListFetchDataSuccess(response)))
-                       .catch((err) => console.log(err))
+                               .catch((err) => {
+                                       // the idea is to redirect to login if something went wrong.
+                                       // TODO: make less wrong
+                                       document.querySelector('a[to="/login"]').click();
+                                       console.log(err)
+                               });
                };
        }